internal/controllers/cluster/cluster_controller_phases_test.go

```diff
@@ -438,25 +438,32 @@ func TestClusterReconcilePhases_reconcileFailureDomains(t *testing.T) {
 			cluster: &clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "test-cluster", Namespace: "test-namespace"}},
 		},
 		{
-			name:     "expect no failure domain if infra config does not have failure domain",
-			cluster:  cluster,
-			infraRef: generateInfraRef(false),
+			name:                 "expect no failure domain if infra config does not have failure domain",
+			cluster:              cluster.DeepCopy(),
+			infraRef:             generateInfraRef(false),
+			expectFailureDomains: clusterv1.FailureDomains{},
+		},
+		{
+			name:                 "expect cluster failure domain to be reset to empty is infra config does not have failure domain",
+			cluster:              clusterWithOldFailureDomain.DeepCopy(),
+			infraRef:             generateInfraRef(false),
+			expectFailureDomains: clusterv1.FailureDomains{},
 		},
 		{
 			name:                 "expect failure domain to remain same if infra config have same failure domain",
-			cluster:              cluster,
+			cluster:              cluster.DeepCopy(),
 			infraRef:             generateInfraRef(true),
 			expectFailureDomains: newFailureDomain,
 		},
 		{
-			name:                 "expect failure domain to be updated if infra config have update to failure domain",
-			cluster:              clusterWithNewFailureDomainUpdated,
+			name:                 "expect failure domain to be updated if infra config has updates to failure domain",
```

Comment:
I think a more appropriate name for `newFailureDomain` would be `infraFailureDomain`.

Also, isn't `expect failure domain to be updated if infra config has updates to failure domain` and `expect failure domain to be reset if infra config have different failure domain` the same test case?